### PR TITLE
Add auto surface extension handler for X11 and Wayland

### PR DESCRIPTION
**Pull request by aka-deVOID**

### CODE REVIEW

File: crates/rhyolite_bevy/Cargo.toml [hunk 1]
```diff
@@ -21,7 +21,8 @@ blocking = "1.2"
 crossbeam-channel = "^0.5"
 tracing = "0.1"
 png = "0.17"
-
+winit = "0.28.6"
```
Comment (Neo-Zhixing) [inline crates/rhyolite_bevy/Cargo.toml:24]: Consider a more relaxed version constraint here. "0.28" would be more than enough so we don't have to update this every time there is a minor version update.

File: crates/rhyolite_bevy/src/lib.rs [hunk 2]
```diff
@@ -100,6 +100,17 @@ impl Plugin for RenderPlugin {
                 .iter()
                 .map(|a| a.as_ptr())
                 .collect();
+
+            let event_loop: &winit::event_loop::EventLoop<()> = app.world.get_non_send_resource().expect("Your display are not supported");
```
Comment (Neo-Zhixing) [inline crates/rhyolite_bevy/src/lib.rs:104]: If we couldn't get the event loop from the world, it probably means that the RenderPlugin was inserted after WinitPlugin. I would just emit a warning with `tracing::warn!()` and skip adding the extension here.

File: crates/rhyolite_bevy/src/lib.rs [hunk 3]
```diff
@@ -100,6 +100,17 @@ impl Plugin for RenderPlugin {
...
+            
+            match event_loop.raw_display_handle() {
+                RawDisplayHandle::Xlib(_) => enabled_instance_extensions.push(ash::extensions::khr::XlibSurface::name().as_ptr()),
+                RawDisplayHandle::Xcb(_) => enabled_instance_extensions.push(ash::extensions::khr::XcbSurface::name().as_ptr()),
+                RawDisplayHandle::Wayland(_) => enabled_instance_extensions.push(ash::extensions::khr::WaylandSurface::name().as_ptr()),
+                RawDisplayHandle::Windows(_) => enabled_instance_extensions.push(ash::extensions::khr::Win32Surface::name().as_ptr()),
+                _ => panic!("Your display are not supported."),
```
Comment (Neo-Zhixing) [inline crates/rhyolite_bevy/src/lib.rs:111]: Same here. Emit a warning and move on.

**Review by Neo-Zhixing — CHANGES_REQUESTED** [review]

File: crates/rhyolite_bevy/Cargo.toml (hunk 1 above)
Comment (bjorn3) [reply crates/rhyolite_bevy/Cargo.toml:24]: A version constraint of `0.28.6` means the same as `>=0.28.6, <0.29`, so cargo still allows updating the patch version.

**Review by aka-deVOID — COMMENTED** [review]
I changed the items.

**Review by Neo-Zhixing — APPROVED** [review]

**Pull request merged**